num_steps: 100
# Landmark parameters
num_landmarks: 50     # number of landmark to sample
space_lim: 20.0   # [meter], we will sample uniformly in [space_lim, space_lim] square

# pose parameters
# We assume that the pose follow a random walk 
sigma_trans: 0.2
sigma_rot: 0.3

odom_noise_trans: 0.3
odom_noise_rot: 0.5

# Landmark parameters
p_obs: 0.2
landmark_noise_trans: 0.1
landmark_noise_rot: 0.1

plot_figure: true


backend:
    #smoother parameter
    type: "isam2" # choose from "fixed_lag", "isam2", "vio" and "none"

    # Load and Save Maps
    prior_map: false
    save_graph: false
    load_path: "" #/home/nvidia/.ros/graph-24-11-2022-15-38-18.g2o"    # prior map
    save_path: ""                 # default: "", system will automatically create file with timestamp  in ~/.ros folder

    # Uncertainty parameters
    landmark_sigma_trans: 0.1   
    landmark_sigma_rot: 0.3
    landmark_prior_sigma_trans: 0.1 
    landmark_prior_sigma_rot: 0.3
    pose_prior_sigma_trans: 0.1
    pose_prior_sigma_rot: 0.3

    # isam2 parameters
    relinearize_threshold: 0.1
    relinearize_skip: 1
    optimizer: "GaussNewton"
    cacheLinearizedFactors: true

    # LM parameter for fixed_lag smoother
    lag: 0.1 # parameter for fix lag only
    lambda_initial: 1e-5
    lambda_upper_bound: 100.0
    lambda_lower_bound: 1e-10
    lambda_factor: 10
    max_iterations: 10
    error_tol: 1e-5
    relative_error_tol: 1e-1
    absolute_error_tol: 1e-3

    # if this is true, we will accept the local optimal solution 
    # when the error no longer decrease rather than increase the lambda
    local_optimal: true 